### PR TITLE
Extract function to compute function selector from signature.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #include <libsolutil/Algorithms.h>
 #include <libsolutil/CommonData.h>
 #include <libsolutil/CommonIO.h>
+#include <libsolutil/FunctionSelector.h>
```
Comment (hrkrshnn) [inline libsolidity/ast/Types.cpp:32]: This file is missing.

Comment (chriseth) [reply libsolidity/ast/Types.cpp:32]: Sorry! Hopefully fixed.

File: libsolutil/FunctionSelector.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#include <libsolutil/Keccak256.h>
+#include <libsolutil/FixedHash.h>
+
+#include <string>
+
+namespace solidity::util
+{
+
+/// @returns the ABI selector for a given function signature, as a 32 bit number.
+inline uint32_t selectorFromSignature32(std::string _signature)
```
Comment (christianparpart) [inline libsolutil/FunctionSelector.h]: Is the parameter being passed as copy by intention? (same question below)

The `keccak256(std::string const&)` function's param is a cref, so I don't see why it was `std::move()`'d. Maybe I am missing something?

Comment (chriseth) [reply libsolutil/FunctionSelector.h]: Fixed.

File: libsolutil/FunctionSelector.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/** @file SHA3.h
```
Comment (leonardoalt) [inline libsolutil/FunctionSelector.h]: Where's this from?

Comment (chriseth) [reply libsolutil/FunctionSelector.h]: *sigh*

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**